index.d.ts

```diff
@@ -0,0 +1,296 @@
+declare module 'react-bulma-components' {
+  import * as React from 'react';
+
+  type Color =
+    | 'primary'
+    | 'success'
+    | 'info'
+    | 'warning'
+    | 'danger'
+    | 'light'
+    | 'dark'
+    | 'white'
+    | 'black'
+    | 'link';
+
+  type Size = 'small' | 'medium' | 'large';
+
+  type Breakpoint =
+    | 'desktop'
+    | 'tablet'
+    | 'mobile'
+    | 'widescreen'
+    | 'fullhd'
+    | 'touch';
+
+  interface HelperProps {
+    clearfix?: boolean;
+    pull?: 'left' | 'right';
+    marginless?: boolean;
+    paddingless?: boolean;
+    overlay?: boolean;
+    clipped?: boolean;
+    radiusless?: boolean;
+    shadowless?: boolean;
+    unselectable?: boolean;
+    invisible?: boolean;
+    hidden?: boolean;
+  }
+
+  interface SizeShape {
+    display?: {
+      value?: 'block' | 'flex' | 'inline' | 'inline-block' | 'inline-flex';
+      only?: boolean;
+    };
+    hide?: {
+      value?: boolean;
+      only?: boolean;
+    };
+    textSize?: {
+      value: 1 | 2 | 3 | 4 | 5 | 6;
+    };
+    textAlignment?: {
+      value?: 'centered' | 'justified' | 'left' | 'right';
+      only?: boolean;
+    };
+  }
+
+  interface ResponsiveProps {
+    responsive?: {
+      mobile?: SizeShape;
+      tablet?: SizeShape;
+      desktop?: SizeShape;
+      widescreen?: SizeShape;
+      fullhd?: SizeShape;
+      touch?: SizeShape;
+    };
+  }
+
+  interface ColorProps {
+    textColor?: string;
+    backgroundColor?: string;
+    colorVariant?: 'light' | 'dark';
+  }
+
+  interface TypographyProps {
+    textSize?: 1 | 2 | 3 | 4 | 5 | 6;
+    textAlignment?: 'centered' | 'justified' | 'left' | 'right';
+    textTransform?: 'capitalized' | 'lowercase' | 'uppercase';
+    textWeight?: 'light' | 'normal' | 'semibold' | 'bold';
+    italic?: boolean;
+  }
+
+  type HTMLAttributes<K extends keyof JSX.IntrinsicElements> = Pick<
+    JSX.IntrinsicElements[K],
+    Exclude<keyof JSX.IntrinsicElements[K], 'unselectable'>
+  >;
+
+  type ModifierProps = HelperProps &
+    ColorProps &
+    ResponsiveProps &
+    TypographyProps;
+
+  type ElementProps = ModifierProps & {
+    className?: string;
+    domRef?: React.RefObject<HTMLElement>;
+  } & (
+      | ({ renderAs?: 'div' } & HTMLAttributes<'div'>)
```

Comment:
Hi @MrCreeper1008 I think it's easier if you accept any string and a React Component, instead of listing every html element. This library is not in charge of validating if the props of a element are correct.

Reply:
Good point. Will fix.

Reply:
@couds I actually managed to provide prop types from `renderAs` prop with a way cleaner solution so I'll keep it that way.